contracts/vesting/VestingEscrowFactory.vy

```diff
@@ -24,80 +38,156 @@ interface VestingEscrowSimple:
 
 event VestingEscrowCreated:
     funder: indexed(address)
-    token: indexed(ERC20)
     recipient: indexed(address)
-    escrow: address
+    index: uint256
     amount: uint256
     vesting_start: uint256
     vesting_duration: uint256
     cliff_length: uint256
-    open_claim: bool
 
+event VestingContractDeployed:
+    recipient: indexed(address)
+    token: indexed(address)
+    index: uint256
+    escrow: address
+    yfi_amount: uint256
+    token_amount: uint256
+
+event LiquidLockerSet:
+    liquid_locker: indexed(address)
+    depositor: address
+
+event OperatorSet:
+    liquid_locker: indexed(address)
+    operator: indexed(address)
+    flag: bool
 
 TARGET: public(immutable(address))
+YFI: public(immutable(ERC20))
+OWNER: public(immutable(address))
 
 
 @external
-def __init__(target: address):
+def __init__(target: address, yfi: address, owner: address):
     """
     @notice Contract constructor
     @dev Prior to deployment you must deploy one copy of `VestingEscrowSimple` which
          is used as a library for vesting contracts deployed by this factory
     @param target `VestingEscrowSimple` contract address
+    @param yfi YFI token address
+    @param owner The owner of the factory that can add/remove liquid lockers and their operators
     """
     TARGET = target
+    YFI = ERC20(yfi)
+    OWNER = owner
 
 @external
-def deploy_vesting_contract(
-    token: ERC20,
-    recipient: address,
-    amount: uint256,
-    vesting_duration: uint256,
+def create_vest(
```

Comment:
ISSUE?

if ychad deposits 100YFI to a recipient and that recipient loses their keys or never starts the vesting deployment the YFI will be locked in this contract forever.

is it worth to add an escape hatch once vesting duration is over to claw back YFI not used for vesting?

Reply:
Yeah, considered this too. Wasn't sure if it was worth it or not, since recipients are heavily incentivized to pick a liquid locker asap. But we might as well add it just in case.